test/TestMLPotential.py

```diff
@@ -1,20 +1,29 @@
 import openmm as mm
+import numpy as np
 import openmm.app as app
 import openmm.unit as unit
 from openmmml import MLPotential
-import unittest
+import pytest
+import itertools
+rtol=1e-5
 
-class TestMLPotential(unittest.TestCase):
 
-    def testCreateMixedSystem(self):
+@pytest.mark.parametrize("implementation,platform_name", list(itertools.product(['nnpops', 'torchani'], ['CUDA', 'CPU', 'Reference', 'OpenCL'])))
+class TestMLPotential:
+
+    def testCreateMixedSystem(self, implementation, platform_name):
         pdb = app.PDBFile('alanine-dipeptide-explicit.pdb')
         ff = app.ForceField('amber14-all.xml', 'amber14/tip3pfb.xml')
         mmSystem = ff.createSystem(pdb.topology, nonbondedMethod=app.PME)
         potential = MLPotential('ani2x')
         mlAtoms = [a.index for a in next(pdb.topology.chains()).atoms()]
-        mixedSystem = potential.createMixedSystem(pdb.topology, mmSystem, mlAtoms, interpolate=False)
-        interpSystem = potential.createMixedSystem(pdb.topology, mmSystem, mlAtoms, interpolate=True)
-        platform = mm.Platform.getPlatformByName('Reference')
+        mixedSystem = potential.createMixedSystem(pdb.topology, mmSystem, mlAtoms, interpolate=False, implementation=implementation)
+        interpSystem = potential.createMixedSystem(pdb.topology, mmSystem, mlAtoms, interpolate=True, implementation=implementation)
+        try:
+            platform = mm.Platform.getPlatformByName(platform_name)
+        except Exception as e:
+            print(f"could not access platform {platform_name} by name; got exception {e}")
+            return
```

Comment:
Instead of hardcoding a list of platforms and returning if one isn't present, let's loop over all available platforms:

```python
for i in range mm.Platform.getNumPlatforms():
    platform = mm.Platform.getPlatform(i)
```

That way we'll automatically pick up future platforms (e.g. HIP) without needing to modify the code.

Reply:
good catch.